### PR TITLE
1583 data layers forest cover rebased

### LINKED ISSUE #1583: Data Layers: Forest Cover
In the data layers add Forest Cover section to toggle visibility of following layers:

- [x] key: JAXA,  label: 'JAXA (2017)'
- [x] key: TandemX,  label: 'TanDEM-X (2019)'
- [x] key: GlobeLand,  label: 'GlobeLand (2020)'
- [x] key: ESAGlobCover ,  label: 'Global Land Cover ESA (2009)'
- [x] key: Copernicus,  label: 'Copernicus (2019)'
- [x] key: ESRI,  label: 'ESRI (2020)'
- [x] key: ESAWorldCover,  label: 'ESA (2020)'
- [x] key: Hansen,  label: 'Hansen GFC (2020)'

### Features to implement:
- [x] get rid of the “accordion” in the title, Forest Cover is the only option here
- [x] Each layer should have a checkbox on the left (see mockup) to toggle its visibility 
- [ ] When ajax call is loading to fetch the mapId, disable the checkbox.

##### endpoint: /api/geo/layers/forest/:countryIso/:forestSource/:gteHansenTreeCoverPerc?
e.g. http://localhost:9000/api/geo/layers/forest/BRA/GlobeLand/

### PR DESCRIPTION
**Pull request by vedran-brnjetic**
Successful merge will close #1583 

Updates

* removed the ui accordion in Forest layers
* added support for forest layers to state and redux
* fixed some minor issues eslint was nagging about (dependency order, useCallback dependency array)
* clicking on checkboxes fetches appropriate layer tiles and marks them checked
* clicking on a checked checkbox removes that layer and clears the checkbox

### CODE REVIEW

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx [hunk 1]
```diff
@@ -1,35 +1,103 @@
...
         {layers.map((layer, index) => (
           <GeoMapMenuListElement
-            key={layer.title}
+            key={layer.key}
             title={layer.title}
             tabIndex={index * -1 - 1}
-            checked={false}
-            onCheckboxClick={() => null}
+            checked={forestOptions.sources.includes(layer.key)}
+            onCheckboxClick={() => onCheckboxClick(layer.apiUri, layer.key, layer.key)}
```
Comment (minotogna) [inline src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx]: you could even define a type layer 
```type Layer = {apiUri:string,key:string,type:string}
const layers:Array<Layer> = ...

// and define 
const onCheckboxClick = useCallback(
     async (layer:Layer) => {
```

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx [hunk 2]
```diff
@@ -1,35 +1,103 @@
...
+    if (overlayLayer.name === mapLayerId) {
+      return true
+    }
+  }
+  return false
+}
+
+const removeOverlayLayer = (mapLayerId: string, overlayLayers: google.maps.MVCArray) => {
+  for (let i = 0; i < overlayLayers.getLength(); i += 1) {
+    const overlayLayer = overlayLayers.getAt(i)
```
Comment (minotogna) [inline src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx]: I saw this loop twice.
maybe better a common method
```
const overlayLayer = findLayer(map,mapLayerId) // I would pass the map reference, not jus the overlayLayers
```

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx (hunk 1 above)
Comment (vedran-brnjetic) [reply src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx]: Yes, actually I started creating a type Layer once I noticed I'll make more use of it, for opacity for instance, and now the palette and who knows. :)

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx (hunk 2 above)
Comment (vedran-brnjetic) [reply src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx]: Yep, noticed it too and made a mental note to optimize, but there's so many mental notes. 
I will certainly refactor a thing or two by next time. I had a goal to make it work, and I hope I didn't cut too many corners in the effort

File: src/meta/geo/forest.ts [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+  | 'JAXA'
+  | 'TandemX'
+  | 'GlobeLand'
+  | 'ESAGlobCover'
+  | 'Copernicus'
+  | 'ESRI'
+  | 'ESAWorldCover'
+  | 'Hansen'
+  | 'MODIS'
+
```
Comment (eraviolo) [inline src/meta/geo/forest.ts]: Maybe is possible to reuse the ForestSource enum values?

Comment (vedran-brnjetic) [reply src/meta/geo/forest.ts]: Yes, I haven't worked with enums much, but it seems to be interchangeable with type for this purpose, so I've done it.

**Comment by vedran-brnjetic** [comment]
image

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**